packages/forge2d/test/dynamics/body_test.dart

```diff
@@ -0,0 +1,56 @@
+import 'package:forge2d/forge2d.dart';
+import 'package:test/expect.dart';
+import 'package:test/scaffolding.dart';
+
+void main() {
+  group('Body', () {
+    group('gravityScale', () {
+      // TODO(alestiago): Make this tests pass.
+      test('scales appropiately in x', () {
+        final gravity = Vector2(10, 0);
+        final world = World(gravity);
+
+        final bodyA = world.createBody(BodyDef()..type = BodyType.dynamic);
+        final bodyB = world.createBody(
+          BodyDef()
+            ..type = BodyType.dynamic
+            ..gravityScale = Vector2(10, 1)
+            ..position = Vector2(0, 10),
+        );
+
+        final bodyAPos = bodyA.position.clone();
+        final bodyBPos = bodyB.position.clone();
+
+        world.stepDt(1);
+
+        expect(bodyAPos.x, isNot(equals(bodyA.position.x)));
+        expect(bodyBPos.x, isNot(equals(bodyB.position.x)));
+
+        expect(bodyB.position.x, greaterThan(bodyA.position.x));
```

Comment:
@spydon I haven't finished these tests. I need to make them pass. If you have suggestions, I'm open.

Going to go to bed now, I might have another look at this tomorrow.

Reply:
Not sure about the current problem, but for many tests in the future you'll probably need an epsilon value that it can diff within both because of float math and the nature of the engine. We have this in some tests in Flame. (On the phone atm so can't send a link, but would be easy enough to find)